### PR TITLE
Add option to use IP address as EFS mount point

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Add option to use IP address as EFS mount point

## Why
* When VPC does not support DNS hostnames we should use IP address for EFS mount target

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -49,7 +49,8 @@ module "elastic_beanstalk_environment" {
   env_vars = "${
       merge(
         map(
-          "EFS_HOST", "${module.efs.dns_name}",
+          "EFS_HOST", "${var.use_efs_ip_address ? module.efs.mount_target_ips[0] : module.efs.dns_name}",
+          "USE_EFS_IP", "${var.use_efs_ip_address}",
```
Comment (comeanother) [inline main.tf]: What is the use case of `USE_EFS_IP` variable in Beanstalk environment?

File: variables.tf [hunk 2]
```diff
@@ -195,3 +195,7 @@ variable "datapipeline_config" {
     timeout       = "60 Minutes"
   }
 }
+
+variable "use_efs_ip_address" {
+  default = false
```
Comment (comeanother) [inline variables.tf]: Use string values. They'll be interpolated as booleans in TF, if boolean is expected.

> Although it appears Terraform supports boolean types, they are instead silently converted to string types. The implications of this are subtle and should be completely understood if you plan on using boolean values.

https://www.terraform.io/docs/configuration/variables.html

File: main.tf (hunk 1 above)
Comment (s2504s) [reply main.tf]: We get value of this ENV variable and using IP address of EFS mount point instead DNS name if `USE_EFS_IP = 1`

**Comment by s2504s** [comment]
image

**Pull request merged**